.circleci/config.yml

```diff
@@ -38,177 +38,232 @@ defaults: &defaults
-                command: /build-tools-ci/scripts/set-environment
-
-            - run:
-                name: log in
-                command: terminus -n auth:login --machine-token="$TERMINUS_TOKEN"
-
-            - run:
-                name: install dev dependencies, build assets, etc.
-                command: ./.ci/scripts/01-prepare
-
-            - run:
-                name: build assets
-                command: composer -n build-assets
-
-            - run:
-                name: prepare database for site-under test
-                command: ./.ci/scripts/02-init-site-under-test-clone-existing
-                # command: ./.ci/scripts/02-init-site-under-test-reinstall-new
-
-            - run:
-                name: run composer install again to get dev dependencies
-                command: composer install
-
-            - run:
-                name: run functional tests with Behat
-                command: ./.ci/scripts/03-test
-
-            - run:
-                name: post-test actions
-                command: ./.ci/scripts/04-post-test
-
-            - run:
-                name: handle merge to master (if needed)
-                command: ./.ci/scripts/05-merge-master
-
-            - run:
-                name: remove transient test fixtures
-                command: ./.ci/scripts/09-cleanup-fixtures
-
-    composer_lock_updater:
-        <<: *defaults
-        docker:
-            - image: quay.io/pantheon-public/build-tools-ci:5.x
-        steps:
-            - checkout
-
-            - run:
-                name: setup-environment-vars
-                command: /build-tools-ci/scripts/set-environment
-
-            - run:
-                name: run composer lock updater
-                command: set -e && clu https://${GITHUB_TOKEN}:x-oauth-basic@github.com/${CI_PROJECT_USERNAME}/${CI_PROJECT_REPONAME}.git | sed -e "s/${GITHUB_TOKEN}/[REDACTED]/g"
-
-    visual_regression_test:
-        <<: *defaults
-        docker:
-            - image: backstopjs/backstopjs:3.9.5
-        steps:
-            - checkout
-
-            - attach_workspace:
-                at: /tmp/workspace
-
-            - run: cp /tmp/workspace/bash_env.txt $BASH_ENV
-            - run: echo "export CI_BUILD_URL='${CIRCLE_BUILD_URL}'" >> $BASH_ENV
-            - run: echo "export CI_NODE_INDEX='${CIRCLE_NODE_INDEX}'" >> $BASH_ENV
-            - run: source $BASH_ENV
-
-            - run:
-                name: visual regression test
-                command: ./.ci/visual-regression-test.sh
-
-            - store_artifacts:
-                path: /tmp/artifacts
-                destination: artifacts
+  configure_env_vars:
+    <<: *defaults
+
+    steps:
+      - checkout
+
+      - run:
+          # Set TERMINUS_ENV and related environment variables.
+          # https://github.com/pantheon-systems/docker-build-tools-ci/blob/6.x/scripts/set-environment
+          name: setup-environment-vars
+          command: /build-tools-ci/scripts/set-environment
```

Comment:
I recall that the `set-environment` script calls `terminus auth:login` as a side-effect. I am now regretting that decision, as not every Circle CI step that calls set-environment necessarily needs to log in. Also, if we save/restore bash_env, then steps that need the terminus login won't get it.

We also made it the responsibility of the set_env script to re-install the Terminus Build Tools plugin if requested. I think that only the Terminus Build Tools plugin project itself needs that.

We'd need to make a 7.x version of the docker container to fix those problems, though.

Reply:
> We'd need to make a 7.x version of the docker container to fix those problems, though.

Agreed, I've opened issues [`32`](https://github.com/pantheon-systems/docker-build-tools-ci/issues/32) and [`33`](https://github.com/pantheon-systems/docker-build-tools-ci/issues/33) on that repository